src/compositions/useStatePolling.ts

```diff
@@ -0,0 +1,33 @@
+import { SubscriptionOptions } from '@prefecthq/vue-compositions'
+import { ComputedRef, computed, ref, watchEffect } from 'vue'
+import { MaybeRef, PrefectStateNames } from '@/types'
+
+type StateResource = {
+  stateName: string | null,
+}
+
+const stateNamesThatShouldPoll: (string | null)[] = ['Pending', 'Running'] satisfies PrefectStateNames[]
+
+export function useStatePolling(stateName: MaybeRef<string | null>, interval?: number): ComputedRef<SubscriptionOptions>
+export function useStatePolling(resource: MaybeRef<StateResource>, interval?: number): ComputedRef<SubscriptionOptions>
+export function useStatePolling(resourceOrStateName: MaybeRef<StateResource | string | null>, interval: number = 5000): ComputedRef<SubscriptionOptions> {
```

Comment:
Just so I understand, the `StateResource` type is so we can directly pass `flowRun` and `taskRun` instances to the composition, right?

Reply:
Correct. That way we don't have to make a computed just to grab the state name from the flow run